pkg/test/compute/cancel_job_test.go

```diff
@@ -0,0 +1,119 @@
+//go:build integration || !unit
+
+package compute
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/devstack"
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	dockermodels "github.com/bacalhau-project/bacalhau/pkg/executor/docker/models"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/node"
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	clientv2 "github.com/bacalhau-project/bacalhau/pkg/publicapi/client/v2"
+	"github.com/bacalhau-project/bacalhau/pkg/setup"
+	"github.com/bacalhau-project/bacalhau/pkg/test/mock"
+	"github.com/bacalhau-project/bacalhau/pkg/test/scenario"
+	"github.com/bacalhau-project/bacalhau/pkg/test/teststack"
+	nodeutils "github.com/bacalhau-project/bacalhau/pkg/test/utils/node"
+	"github.com/stretchr/testify/suite"
+)
+
+type CancelJobTestSuite struct {
+	suite.Suite
+}
+
+func TestCancelJobTestSuite(t *testing.T) {
+	suite.Run(t, new(CancelJobTestSuite))
+}
+
+func (s *CancelJobTestSuite) TestJobCancellation() {
+	ctx := context.Background()
+
+	docker.MustHaveDocker(s.T())
+
+	fsr, c := setup.SetupBacalhauRepoForTesting(s.T())
+
+	computeConfig, err := node.NewComputeConfigWith(c.Node.ComputeStoragePath, node.ComputeConfigParams{
+		TotalResourceLimits: models.Resources{
+			CPU:    1,
+			Memory: 1 * 1024 * 1024 * 1024,
+			Disk:   1 * 1024 * 1024 * 1024,
+		},
+		IgnorePhysicalResourceLimits: true,
+		ControlPlaneSettings: types.ComputeControlPlaneConfig{
+			ResourceUpdateFrequency: types.Duration(50 * time.Millisecond),
+		},
+	})
+
+	s.Require().NoError(err)
+
+	requesterConfig, err := node.NewRequesterConfigWith(node.RequesterConfigParams{
+		NodeOverSubscriptionFactor: 2,
+	})
+	s.Require().NoError(err)
+
+	stack := teststack.Setup(ctx, s.T(), fsr, c,
+		devstack.WithNumberOfRequesterOnlyNodes(1),
+		devstack.WithNumberOfComputeOnlyNodes(1),
+		devstack.WithComputeConfig(computeConfig),
+		devstack.WithRequesterConfig(requesterConfig),
+	)
+
+	nodeutils.WaitForNodeDiscovery(s.T(), stack.Nodes[0].RequesterNode, 2)
+
+	es, err := dockermodels.NewDockerEngineBuilder("ubuntu").
+		WithEntrypoint("bash", "-c", "sleep 100000").
+		Build()
+
+	s.Require().NoError(err)
+	task := mock.TaskBuilder().
+		Engine(es).
+		BuildOrDie()
+	job := mock.Job()
+	job.Tasks[0] = task
+
+	job.Normalize()
+
+	apiServer := stack.Nodes[0].APIServer
+	apiProtocol := "http"
+	apiHost := apiServer.Address
+	apiPort := apiServer.Port
+	api := clientv2.New(fmt.Sprintf("%s://%s:%d", apiProtocol, apiHost, apiPort))
+
+	submittedJob, err := api.Jobs().Put(ctx, &apimodels.PutJobRequest{
+		Job: job,
+	})
+	s.Require().NoError(err)
+
+	resolver := scenario.NewStateResolverFromAPI(api)
+	time.Sleep(time.Second * 10)
```

Comment:
We don't want to slow our test suites more than how slow they are already. You may want to look into `noop_executor` instead of docker where you can inject a job handler. You can use and block on a channel and have the job handler add a value to the channel to indicate the execution started, and then sleep

Reply:
I can't do that right. The code changes are part of docker execution engine. If i don't use docker, i won't be testing the right thing. Here we are making changes to specific engine. 

Also my bad, i don't need that sleep there ! Will get rid of it !